src/i18n/hooks/use-dictionary.tsx

```diff
@@ -0,0 +1,72 @@
+// use-dictionary.ts
+"use client";
+
+import { useState, useEffect, createContext, useContext } from "react";
+import { getDictionary } from "@/i18n/dictionary";
+import { useLocale } from "@/i18n/hooks/use-locale";
+import type { Dictionary } from "@/i18n/types";
+import type { Locale } from "@/i18n/config";
+
+type DictionaryContextType = {
+	dictionary: Dictionary | undefined;
```

Comment:
Switching this to simply dict per other suggestions.
```suggestion
	dict: Dictionary | undefined;
```